src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs

```diff
@@ -28,6 +28,9 @@ public partial class Socket
         /// <summary>Cached instance for send operations that return <see cref="Task{Int32}"/>.</summary>
         private TaskSocketAsyncEventArgs<int>? _multiBufferSendEventArgs;
 
+        /// <summary>Cached instance for file send operations that return <see cref="ValueTask"/>.</summary>
+        private FileSendSocketAsyncEventargs? _fileSendEventArgs;
```

Comment:
Can we just use one of the existing send instances (above) here?

Reply:
In the current state of the PR no, as FileSendSocketAsyncEventargs is special to windows and potential parent classes are sealed. Unsealing them might change perf-characteristics or would this be OK?

But I'll try something to unifiy this. Adding the fields to `SocketAsyncEventArgs` and use `AwaitableSocketAsyncEventArgs`, then one field could be re-used.